### ISSUE TITLE
Bloated git repository

### ISSUE THREAD

**Issue by billsacks** [ESMCI/cime#2458]
I noticed that recent clones of cime are much bigger than they used to be. I tracked the problem back to 92ccb8380 - `Merge pull request #2312 from ESMCI/jgfouca/branch-for-acme-split-2018-02-22` - which added 5,296 commits to history. https://github.com/ESMCI/cime/pull/2357 also added a lot of commits, and there was a brief discussion in that PR about that issue. It looks like the two most recent e3sm splits (https://github.com/ESMCI/cime/pull/2406 and https://github.com/ESMCI/cime/pull/2433) only added a small number of commits. Maybe the changes with https://github.com/ESMCI/cime/pull/2367 fixed this?

There probably isn't much we can do about this at this point, but I wanted to make sure that the e3sm split process will involve relatively few added commits moving forward so we don't experience runaway repository bloat. @jgfouca ?

**Comment by billsacks** [comment]
In case it's of interest, here's what I found in terms of repository growth over the last couple of years. These are the data sizes pulled down with `git clone -b TAGNAME --single-branch git@github.com:ESMCI/cime.git`

4.6.0 was 13.55 MiB

5.0.0 was 17.79 MiB

5.4.0-alpha01 was 26.98 MiB

5.4.0-alpha16 was 29.35 MiB

5.4.0-alpha22 was 30.33 MiB

5.4.0-alpha24 was 30.50 MiB

5.4.0-alpha25 was 69.55 MiB

5.4.0-alpha26 was 70.47 MiB

master is 70.71 MiB

**Comment by jgfouca** [comment]
@billsacks , thanks for this investigation. I will keep an eye on the acme splits to make sure they aren't adding too many commits. Maybe we should go back to squashing to reduce history bloat?

**Comment by rljacob** [comment]
Yes do squash commits each way.

**Comment by jgfouca** [comment]
I think @billsacks asked for the full history at one point.

**Comment by billsacks** [comment]
To clarify, I said that squashing didn't seem ideal, but I was okay with it (https://github.com/ESMCI/cime/issues/2177#issuecomment-356777433 and https://github.com/ESMCI/cime/issues/2177#issuecomment-366413324).

Based on a few spot-checks from the last two e3sm split PRs, it looks like the only commits being added to history are ones that actually touched cime – which seems like the right behavior. I'd be concerned if somehow all of the e3sm commits were being added to history, and/or if the number of commits coming from the e3sm splits were, say, an order of magnitude larger than the number of commits being added directly to cime.

To summarize: I'm fine with the status quo as long as you keep an eye on this, like you suggest @jgfouca . But I'm also fine with having you squash them if you prefer that.

**Comment by billsacks** [comment]
I'll go ahead and close this because I think we've resolved it enough; feel free to reopen if you want to discuss further.

**Comment by billsacks** [comment]
This issue of repository bloat came up again on today's CIME call.

Current repository size:
- Full repository: 85.59 MiB
- `--single-branch master`: 80.16 MiB (growth of a little less than 10 MiB since April, 2018).

@jgfouca raised the idea of, at some point, cutting off history older than some point and force pushing to master. (We could still keep the old history in an archived repo somewhere.) We didn't decide if this is worth doing; we can revisit this later.

**Comment by jgfouca** [comment]
Just a little additional info, as of today, it takes 12s to clone the repo on a local file system with a fast network.
